sdk/servicebus/azure-servicebus/azure/servicebus/aio/async_message.py

```diff
@@ -4,156 +4,29 @@
-        :raises: ~azure.servicebus.common.errors.MessageSettleFailed if message settle operation fails.
-        """
-        self._is_live('reject')
-        try:
-            reject = functools.partial(self.message.reject, condition=DEADLETTERNAME, description=description)
-            await self._loop.run_in_executor(None, reject)
-        except Exception as e:
-            raise MessageSettleFailed("reject", e)
-
-    async def abandon(self):
-        """Abandon the message.
-
-        This message will be returned to the queue to be reprocessed.
-
-        :raises: ~azure.servicebus.common.errors.MessageAlreadySettled if the message has been settled.
-        :raises: ~azure.servicebus.common.errors.MessageLockExpired if message lock has already expired.
-        :raises: ~azure.servicebus.common.errors.SessionLockExpired if session lock has already expired.
-        :raises: ~azure.servicebus.common.errors.MessageSettleFailed if message settle operation fails.
-        """
-        self._is_live('abandon')
-        try:
-            modify = functools.partial(self.message.modify, True, False)
-            await self._loop.run_in_executor(None, modify)
-        except Exception as e:
-            raise MessageSettleFailed("abandon", e)
-
-    async def defer(self):
-        """Defer the message.
-
-        This message will remain in the queue but must be received
-        specifically by its sequence number in order to be processed.
-
-        :raises: ~azure.servicebus.common.errors.MessageAlreadySettled if the message has been settled.
-        :raises: ~azure.servicebus.common.errors.MessageLockExpired if message lock has already expired.
-        :raises: ~azure.servicebus.common.errors.SessionLockExpired if session lock has already expired.
-        :raises: ~azure.servicebus.common.errors.MessageSettleFailed if message settle operation fails.
-        """
-        self._is_live('defer')
-        try:
-            modify = functools.partial(self.message.modify, True, True)
-            await self._loop.run_in_executor(None, modify)
-        except Exception as e:
-            raise MessageSettleFailed("defer", e)
-
-
-class DeferredMessage(Message):
-    """A message that has been deferred.
-
-    A deferred message can be completed,
-    abandoned, or dead-lettered, however it cannot be deferred again.
     """
 
-    def __init__(self, deferred, mode):
-        self._settled = mode == 0
-        super(DeferredMessage, self).__init__(None, message=deferred)
-
-    def _is_live(self, action):
-        if not self._receiver:
-            raise ValueError("Orphan message had no open connection.")
-        super(DeferredMessage, self)._is_live(action)
+    def __init__(self, message, mode=ReceiveSettleMode.PeekLock, loop=None):
+        self._loop = loop or get_running_loop()
+        super(ReceivedMessage, self).__init__(message=message, mode=mode)
 
     @property
     def lock_token(self):
```

Comment:
This isn't defined on the child class?

Reply:
my bad, this can be removed, thanks for pointing out.